libraries/hotstuff/include/eosio/hotstuff/qc_chain.hpp

```diff
@@ -38,28 +38,34 @@ namespace eosio::hotstuff {
 
       explicit quorum_certificate(const quorum_certificate_message& msg)
               : proposal_id(msg.proposal_id)
-              , active_finalizers(msg.active_finalizers) // conversion from string
+              , active_finalizers()
```

Comment:
minor... this line is not needed.
Why not initialize directly from the message:
`, active_finalizers(msg.active_finalizers.cbegin(), msg.active_finalizers.cend())`